bin/cli.js

```diff
@@ -55,12 +55,18 @@ function mkConfigObj(opts) {
 function initKnex(env, opts) {
   checkLocalModule(env);
 
-  if (!env.configuration) {
-    if (opts.client) env.configuration = mkConfigObj(opts);
-    else
+  if (!env.pathToKnexFile) {
```

Comment:
I've separated `pathToKnexFile` and `configuration` properties to reduce confusion.